src/State/OrganisationTreeProvider.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\Exception\ProviderException;
+use App\Repository\OrganisationDataRepository;
+
+readonly class OrganisationTreeProvider implements ProviderInterface
+{
+    public function __construct(private OrganisationDataRepository $organisationDataRepository)
+    {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        if (!isset($uriVariables['id'])) {
+            throw new ProviderException('Could not find id in uri');
+        }
+
+        $id = $uriVariables['id'];
+
+        $organisationer = [];
+
+        // Traverse through enheder until top is reached.
+        while (true) {
+            $organisation = $this->organisationDataRepository->findOneBy(['id' => $id]);
+
+            if (!$organisation) {
+                throw new ProviderException(sprintf('Could not find organisation with id %s', $id));
```

Comment:
The caller has provided a single ID and telling that another ID is invalid may be confusing. Consider rewording the exception message to something like “Error build organization path (parent organization with ID %s does not exist)”.

Reply:
Fair point. Reworded exception message.